holohub

```diff
@@ -0,0 +1,1022 @@
+        if args.local or os.environ.get("HOLOHUB_BUILD_LOCAL"):
+            project_data = self._find_project(
+                project_name=args.project,
+                language=args.language if hasattr(args, "language") else None,
+            )
+            if not project_data:
+                holohub_cli_util.fatal(
+                    f"Project '{args.project}' {f'({args.language})' if hasattr(args, 'language') else ''} not found"
+                )
+
+            build_dir = HoloHubCLI.DEFAULT_BUILD_PARENT_DIR / args.project
+            BUILD_TYPE = "Release"
+            holohub_cli_util.run_command(
+                [
+                    "cmake",
+                    "-B",
+                    str(build_dir),
+                    "-S",
+                    str(HoloHubCLI.HOLOHUB_ROOT),
+                    "-G",
+                    "Ninja",
+                    f"-DCMAKE_BUILD_TYPE={BUILD_TYPE}",
+                    f"-DCMAKE_PREFIX_PATH={HoloHubCLI.DEFAULT_SDK_DIR}",
+                    f"-DAPP_{args.project}=ON",
+                ],
+                dry_run=args.dryrun,
+            )
+            holohub_cli_util.run_command(
+                ["cmake", "--build", str(build_dir), "--config", BUILD_TYPE], dry_run=args.dryrun
+            )
+
+            language = holohub_cli_util.normalize_language(
+                project_data.get("metadata", {}).get("language", None)
+            )
+
+            run_config = project_data.get("metadata", {}).get("run", {})
+            if not run_config:
+                holohub_cli_util.fatal(
+                    f"Project '{args.project}' does not have a run configuration"
+                )
+
+            app_source_path = project_data.get("source_folder", "")
+
+            # Process command template
+            cmd = run_config["command"]
+            cmd = cmd.replace("<holohub_data_dir>", str(HoloHubCLI.DEFAULT_DATA_DIR))
+            cmd = cmd.replace("<holohub_app_source>", str(app_source_path))
+
+            cmd = cmd.replace("<holohub_bin>", str(build_dir))
+
+            app_build_dir = build_dir / app_source_path.relative_to(HoloHubCLI.HOLOHUB_ROOT)
+            cmd = cmd.replace("<holohub_app_bin>", str(app_build_dir))
+
+            if language == "cpp":
+                if not build_dir.is_dir() and not args.dryrun:
+                    holohub_cli_util.fatal(
+                        f"The build directory for this application does not exist.\n"
+                        f"Did you forget to './holohub build {args.project}'?"
+                    )
+
+            # Handle workdir
+            workdir = run_config.get("workdir", "holohub_app_bin")
+            if workdir == "holohub_app_source":
+                if not args.dryrun:
+                    os.chdir(project_data.get("source_folder", ""))
+                print(
+                    f"\033[34m{holohub_cli_util.get_timestamp()} \033[37m$ \033[32mcd {project_data.get('source_folder', '')}\033[0m"
```

Comment:
the color rendering could be unified and decoupled to a separate class

Reply:
Given the limited scope I'd like to defer on this for now, but open to revisiting in the future.